osquery/filesystem/fileops.h

```diff
@@ -171,10 +172,20 @@ class PlatformFile {
    */
   Status isOwnerCurrentUser() const;
 
-  /// Determines whether the file has the executable bit set (currently returns
-  /// true)
+  /// Determines whether the file has the executable bit set 
   Status isExecutable() const;
 
+  /**
+   * @brief Determines how "safe the executable it is from external modification
+   * @note Currently, this is only implemented on Windows. The Windows version of
+   *       this function ensures that writes are explicitly denied for the file and
+   *       the file's parent directory.
+   */
+  Status isSafeForLoading() const;
+
+  /// Determines whether the file is a directory
+  Status isDirectory() const;
```

Comment:
Do you want to replace the directory check in `safePermissions` with the `filesystem.cpp` `isDirectory`?